.github/workflows/run_tests.yml

```diff
@@ -0,0 +1,71 @@
+name: test
+
+on:
+  push:
+    branches:
+      - "main"
+      - "master"
+      - "dev"
+    paths:
+      - "**.py"
+      - ".github/workflows/*test*.yml"
+      - "pyproject.toml"
+      - "poetry.lock"
+      - "requirements/requirements*.txt"
+      - "requirements/requirements-dask.txt"
+  pull_request:
+    paths:
+      - "**.py"
+      - ".github/workflows/*test*.yml"
+      - "pyproject.toml"
+      - "poetry.lock"
+      - "requirements/requirements*.txt"
+      - "requirements/requirements-dask.txt"
+
+jobs:
+  test:
+    name: "test Python ${{ matrix.python }} on ${{ matrix.os }} with Dask ${{ matrix.dask }}"
+    runs-on: "${{ matrix.os }}"
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest"]
+        python: ["3.10", "3.11"]
+        dask: [true, false]
+        include:
+          - os: "macos-latest"
+            python-version: "3.10"
+            dask: false
+    defaults:
+      run:
+        shell: "bash -elo pipefail {0}"
```

Comment:
Remove the `-l`. It seems that login shells on github actions are not compatible with MacOS. See here for details: https://github.com/actions/setup-python/issues/132#issuecomment-757752092